src/main/java/at/ac/tuwien/kr/alpha/common/Symbol.java

```diff
@@ -9,22 +9,25 @@
  * Copyright (c) 2016, the Alpha Team.
  */
 public class Symbol implements Comparable<Symbol> {
+	private static int counter;
 	private static final Interner<Symbol> INTERNER = Interners.newStrongInterner();
 
 	private final String symbol;
 	private final int arity;
+	private final int id;
 
-	private Symbol(String symbol, int arity) {
+	private Symbol(String symbol, int arity, int id) {
 		this.symbol = symbol;
 		this.arity = arity;
+		this.id = id;
 	}
 
 	public static Symbol getInstance(String symbol) {
 		return getInstance(symbol, 0);
 	}
 
 	public static Symbol getInstance(String symbol, int arity) {
-		return INTERNER.intern(new Symbol(symbol, arity));
+		return INTERNER.intern(new Symbol(symbol, arity, counter++));
```

Comment:
Does this mean that `Symbol.getInstance("a",0) != Symbol.getInstance("a",0)` ? It looks wrong to me, why is this okay?